includes/checkout/template.php

```diff
@@ -1099,5 +1099,23 @@ function edd_receipt_show_download_files( $item_id, $receipt_args, $item = array
 		}
 	}
 
-	return apply_filters( 'edd_receipt_show_download_files', $ret, $item_id, $receipt_args, $item );
+	if ( has_filter( 'edd_receipt_show_download_files' ) ) {
+		$cart = array();
+		if ( ! empty( $item->order_id ) ) {
+			$order = edd_get_order_by( 'id', $item->order_id );
+			$cart  = edd_get_payment_meta_cart_details( $order->id, true );
+		}
+		$ret = apply_filters( 'edd_receipt_show_download_files', $ret, $item_id, $receipt_args, $cart[ $item->cart_index ] );
```

Comment:
I think we need to work in some backwards compatibility logic here. I see that you're now passing in an `Order_Item` object to `edd_receipt_show_download_files()`, but previously `$item` was an array. So someone using an old template could still be passing one in. We need to account for that.

So if `$item` is an array, then that should be passed in as the last argument to this hook. If it's an `Order_Item` object, then we have to run this backwards compat logic.

Reply:
Also, can you update the docblock for `edd_receipt_show_download_files()` to clarify that `$item` can now also be an `Order_Item` object?